### PR TITLE
Leaderboard v3

### PR DESCRIPTION
**Pull request by phelios**
Updates:

- Gone a little wild with refactoring 😬 
- Add titles to leaderboards
- Default leaderboard to weekly

image

### CODE REVIEW

File: emojirades/commands/scorekeeper_commands/leaderboard_command.py [hunk 1]
```diff
@@ -1,73 +1,48 @@
...
 
 
 class LeaderboardCommand(BaseCommand):
+
     description = "Shows all the users scores"
 
     patterns = (
         r"<@{me}> (score|leader)[\s]*board$",
-        r"<@{me}> (score|leader)[\s]*board (?P<range>weekly|monthly)",
+        r"<@{me}> (score|leader)[\s]*board (?P<range>weekly|monthly|alltime)",
```
Comment (michael-robbins) [inline emojirades/commands/scorekeeper_commands/leaderboard_command.py:15]: I kinda like 'all time' as 'alltime' seems a bit weird, as it's not a single word? Can we cater for both?

File: emojirades/commands/scorekeeper_commands/leaderboard_command.py [hunk 2]
```diff
@@ -1,73 +1,48 @@
...
     examples = [
         ("<@{me}> scoreboard", "Show user scores"),
+        ("<@{me}> scoreboard weekly|monthly|alltime", "Show user scores on different brackets"),
+        ("<@{me}> leaderboard", "Alternative name for scoreboard"),
     ]
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        # TODO: :thinking: Maybe we could add default value regex in the command translation
```
Comment (michael-robbins) [inline emojirades/commands/scorekeeper_commands/leaderboard_command.py:27]: That sounds too crazy, I like this way!

File: emojirades/commands/scorekeeper_commands/leaderboard_command.py [hunk 3]
```diff
@@ -1,73 +1,48 @@
...
+        self.time_unit = self.args.get("range", TimeRange.WEEKLY)
+
     def execute(self):
         yield from super().execute()
         leaderboard = []
 
-        time_unit = self.args.get("range")
-        if time_unit:
-            self.logger.debug(f"Getting a {time_unit} leaderboard")
+        if self.time_unit == TimeRange.ALLTIME:
```
Comment (michael-robbins) [inline emojirades/commands/scorekeeper_commands/leaderboard_command.py:34]: Same deal here TimeRange.ALL_TIME sounds better to me, thoughts?

File: emojirades/printers/leaderboard_printer.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    def print(self):
+        leaderboard = self.data
+
+        self.logger.debug(f"Printing leaderboard: {leaderboard}")
+
+        if not leaderboard:
+            yield None, "Nothing to see here!"
+            return
+
+        lines = ["```", f" :: {self.time_unit.capitalize()} leaderboard ::", ""]
```
Comment (michael-robbins) [inline emojirades/printers/leaderboard_printer.py:20]: And if Alltime as printed here, it'd be `All Time`?

**Review by michael-robbins — CHANGES_REQUESTED** [review]
Thoughts on the alltime? I'm 100% fine with the rest though, lookin pretty 🔥

**Comment by phelios** [comment]
@michael-robbins totally with you on this. I was on the fence about reverting it back to "overall". I just don't have something to compare to. But Happy with ALL_TIME and "All Time". Will fix that tomorrow, not on my workstation anymore :)

**Review by michael-robbins — APPROVED** [review]

**Pull request merged**